preflight-packages/examples.jetstack.io/pods_basic/pods.rego

```diff
@@ -3,67 +3,74 @@ package pods
-        count(init_containers_without_memory_limits) == 0
-}
-memory_limit_unset(container) {
-        not container.resources.limits.memory
+preflight_container_mem_limit[message] {
+	# find all containers in all pods
+	pod := pods.items[_]
+	container := pod.spec.containers[_]
+	# test if the limits are not set
+	not container.resources.limits.memory
+	# bind a message for reporting
+	message := sprintf("container '%s' in pod '%s' in namespace '%s' is missing a memory limit", [container.name, pod.metadata.name, pod.metadata.namespace])
 }
-containers_without_memory_limits[container_name] {
-        pod := pods.items[_]
-        container := pod.spec.containers[_]
-        container_name = format_container(pod, container)
-        memory_limit_unset(container)
-}
-init_containers_without_memory_limits[container_name] {
-        pod := pods.items[_]
-        container := pod.spec.initContainers[_]
-        container_name = format_container(pod, container)
-        memory_limit_unset(container)
+preflight_container_mem_limit[message] {
+	# find all initContainers in all pods
+	pod := pods.items[_]
+	container := pod.spec.initContainers[_]
+	# test if the limits are not set
+	not container.resources.limits.memory
+	# bind a message for reporting
+	message := sprintf("init container '%s' in pod '%s' in namespace '%s' is missing a memory limit", [container.name, pod.metadata.name, pod.metadata.namespace])
 }
 
 # Rule 'explicit_image_tag'
-default preflight_explicit_image_tag = false
-preflight_explicit_image_tag {
-        count(containers_without_explicit_tag) == 0
+preflight_explicit_image_tag[message] {
+	# find all containers in all pods
+	pod := pods.items[_]
+	container := pod.spec.containers[_]
+	# validate that the image value contains an explicit tag
+	{ re_match(`latest$`, container.image),
+	re_match(`^[^:]+$`, container.image) } & { true } != set()
+
+	# bind a message for reporting
+	message := sprintf("container '%s' in pod '%s' in namespace '%s' is missing an explicit image tag", [container.name, pod.metadata.name, pod.metadata.namespace])
 }
-explicit_tag(container) {
-        not re_match(`^.*:latest$`, container.image)
-        re_match(`^.*:.*$`, container.image)
+
+preflight_explicit_image_tag[message] {
+	# find all containers in all pods
+	pod := pods.items[_]
+	container := pod.spec.initContainers[_]
+	# validate that the image value contains an explicit tag
+	{ re_match(`latest$`, container.image),
+	re_match(`^[^:]+$`, container.image) } & { true } != set()
+
+	# bind a message for reporting
+	message := sprintf("init container '%s' in pod '%s' in namespace '%s' is missing an explicit image tag", [container.name, pod.metadata.name, pod.metadata.namespace])
 }
-containers_without_explicit_tag[container_name] {
-        pod := pods.items[_]
-        container := pod.spec.containers[_]
-        container_name = format_container(pod, container)
-        not explicit_tag(container)
-}
```

Comment:
No newline at end of file.

Reply:
I think there is now though right?

Reply:
This is the old content.

Reply:
Ah yeah, GitHub diff is saying you've removed the lack of newline at the end of the file. So all good!